AnnoDesigner/Models/LayoutObject.cs

```diff
@@ -194,6 +252,42 @@ private Rect ScreenRect
             }
         }
 
+        public Rect? CalculateReservedScreenRect(int gridSize)
+        {
```

Comment:
Is it possible to "cache" the calculated rectangle? I also don't know if it would be beneficial, but the result of `CalculateScreenRect` is also cached and it improved the rendering performance a bit.

Reply:
Sure, but looking at the amount of stuff being cached in LayoutObject, it also might be considered a problem :sweat_smile:

Reply:
Caching for the other stuff was added after some profiling sessions to improve the rendering performance (due to way lesser pressure on the GC).